### PR TITLE
Free space monitoring and crash dump uploading

### PR DESCRIPTION
**Pull request by ChadNedzlek**

### CODE REVIEW

File: src/Maestro/CoreHealthMonitor/.config/settings.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "HealthTableUri": "{vault(health-report-table-sas-uri)]",
```
Comment (garath) [inline src/Maestro/CoreHealthMonitor/.config/settings.json:2]:
```suggestion
  "HealthTableUri": "[vault(health-report-table-sas-uri)]",
```

File: src/Maestro/DependencyUpdateErrorProcessor/.config/settings.json [hunk 2]
```diff
@@ -1,4 +1,5 @@
 {
+  "HealthTableUri": "{vault(health-report-table-sas-uri)]",
```
Comment (garath) [inline src/Maestro/DependencyUpdateErrorProcessor/.config/settings.json:2]:
```suggestion
  "HealthTableUri": "[vault(health-report-table-sas-uri)]",
```

File: src/Maestro/CoreHealthMonitor/CoreHealthMonitor.cs [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        {
+            foreach (DriveInfo drive in DriveInfo.GetDrives())
+            {
+                long threshold = _driveOptions.Value.MinimumFreeSpaceBytes;
+                long freeSpace = drive.AvailableFreeSpace;
+
+                if (freeSpace < threshold)
+                {
+                    _logger.LogInformation(
+                        "Available drive space on {drive} is at {freeSpace} above threshold of {threshold} bytes",
```
Comment (alexperovich) [inline src/Maestro/CoreHealthMonitor/CoreHealthMonitor.cs:72]:
```suggestion
                        "Available drive space on {drive} is at {freeSpace} below threshold of {threshold} bytes",
```

File: src/Maestro/SubscriptionActorService/.config/settings.json [hunk 4]
```diff
@@ -1,4 +1,5 @@
 {
+  "HealthTableUri": "{vault(health-report-table-sas-uri)]",
```
Comment (garath) [inline src/Maestro/SubscriptionActorService/.config/settings.json:2]:
```suggestion
  "HealthTableUri": "[vault(health-report-table-sas-uri)]",
```

File: src/Maestro/CoreHealthMonitor/CoreHealthMonitor.cs (hunk 3 above)
Comment (alexperovich) [reply src/Maestro/CoreHealthMonitor/CoreHealthMonitor.cs:72]: I think you just swapped the log statements. The error one should be here.

File: src/Maestro/DependencyUpdateErrorProcessor/Program.cs [hunk 5]
```diff
@@ -48,18 +50,21 @@ public static void Configure(IServiceCollection services)
             services.Configure<DependencyUpdateErrorProcessorOptions>(
                 (options, provider) =>
                 {
-                    var config = provider.GetRequiredService<IConfiguration>();
-                    if (!bool.TryParse(config["EnableDependencyUpdateErrorProcessor"], out var errorProcessorFlag))
+                    var config1 = provider.GetRequiredService<IConfiguration>();
```
Comment (alexperovich) [inline src/Maestro/DependencyUpdateErrorProcessor/Program.cs]: why the `1`?

File: src/Maestro/DependencyUpdater/Program.cs [hunk 6]
```diff
@@ -50,18 +49,16 @@ public static void Configure(IServiceCollection services)
...
-            services.Configure<GitHubTokenProviderOptions>(
-                (options, provider) =>
-                {
-                    var config = provider.GetRequiredService<IConfiguration>();
-                    IConfigurationSection section = config.GetSection("GitHub");
-                    section.Bind(options);
-                });
+            services.Configure<GitHubTokenProviderOptions>((options, provider) =>
+            {
+                var config1 = provider.GetRequiredService<IConfiguration>();
```
Comment (alexperovich) [inline src/Maestro/DependencyUpdater/Program.cs:54]: its here too?

File: src/Maestro/DependencyUpdateErrorProcessor/Program.cs (hunk 5 above)
Comment (ChadNedzlek) [reply src/Maestro/DependencyUpdateErrorProcessor/Program.cs]: ReSharper did it... Need to fix them all..

File: src/Telemetry/AzureDevOpsTimeline/.config/settings.json [hunk 7]
```diff
@@ -1,17 +1,18 @@
 {
-    "KeyVaultUri": "",
+  "HealthTableUri": "{vault(health-report-table-sas-uri)]",
```
Comment (garath) [inline src/Telemetry/AzureDevOpsTimeline/.config/settings.json:2]:
```suggestion
  "HealthTableUri": "[vault(health-report-table-sas-uri)]",
```

Comment (alexperovich) [inline src/Telemetry/AzureDevOpsTimeline/.config/settings.json:2]:
```suggestion
  "HealthTableUri": "[vault(health-report-table-sas-uri)]",
```

File: src/DotNet.Status.Web/appsettings.json [hunk 8]
```diff
@@ -1,4 +1,5 @@
 {
+  "HealthTableUri": "{vault(health-report-table-sas-uri)]",
```
Comment (garath) [inline src/DotNet.Status.Web/appsettings.json:2]:
```suggestion
  "HealthTableUri": "[vault(health-report-table-sas-uri)]",
```

File: src/Maestro/DependencyUpdater/.config/settings.json [hunk 9]
```diff
@@ -1,4 +1,5 @@
 {
+  "HealthTableUri": "{vault(health-report-table-sas-uri)]",
```
Comment (garath) [inline src/Maestro/DependencyUpdater/.config/settings.json:2]:
```suggestion
  "HealthTableUri": "[vault(health-report-table-sas-uri)]",
```

File: src/Maestro/FeedCleanerService/.config/settings.json [hunk 10]
```diff
@@ -1,4 +1,5 @@
 {
+  "HealthTableUri": "{vault(health-report-table-sas-uri)]",
```
Comment (garath) [inline src/Maestro/FeedCleanerService/.config/settings.json:2]:
```suggestion
  "HealthTableUri": "[vault(health-report-table-sas-uri)]",
```

File: src/Maestro/CoreHealthMonitor/CoreHealthMonitor.cs [hunk 11]
```diff
@@ -0,0 +1,150 @@
...
+        private async Task ScanDriveFreeSpaceAsync()
+        {
+            foreach (DriveInfo drive in DriveInfo.GetDrives())
+            {
+                long threshold = _driveOptions.Value.MinimumFreeSpaceBytes;
+                long freeSpace = drive.AvailableFreeSpace;
+
+                if (freeSpace < threshold)
+                {
+                    _logger.LogInformation(
```
Comment (alexperovich) [inline src/Maestro/CoreHealthMonitor/CoreHealthMonitor.cs:71]:
```suggestion
                    _logger.LogError(
```

File: src/Maestro/CoreHealthMonitor/CoreHealthMonitor.cs [hunk 12]
```diff
@@ -0,0 +1,150 @@
...
+                    await _health.UpdateStatusAsync(
+                            "DriveSpace:" + drive.Name,
+                            HealthStatus.Error,
+                            $"Available drive space on {drive.Name} is at {freeSpace:N} below threshold of {threshold:N} bytes"
+                        )
+                        .ConfigureAwait(false);
+                }
+                else
+                {
+                    _logger.LogError(
```
Comment (alexperovich) [inline src/Maestro/CoreHealthMonitor/CoreHealthMonitor.cs]:
```suggestion
                    _logger.LogInformation(
```

File: src/Maestro/CoreHealthMonitor/CoreHealthMonitor.cs [hunk 13]
```diff
@@ -0,0 +1,150 @@
...
+                        "Available drive space on {drive} is at {freeSpace} below threshold of {threshold} bytes",
```
Comment (alexperovich) [inline src/Maestro/CoreHealthMonitor/CoreHealthMonitor.cs]:
```suggestion
                        "Available drive space on {drive} is at {freeSpace} above threshold of {threshold} bytes",
```

File: src/Maestro/Maestro.Web/.config/settings.json [hunk 14]
```diff
@@ -1,4 +1,5 @@
 {
+  "HealthTableUri": "{vault(health-report-table-sas-uri)]",
```
Comment (garath) [inline src/Maestro/Maestro.Web/.config/settings.json]:
```suggestion
  "HealthTableUri": "[vault(health-report-table-sas-uri)]",
```

**Comment by riarenas** [comment]
@ChadNedzlek Looks like this ended up breaking the Maestro SF deployment:
```
Overriding application parameter file specified in publish profile with 'D:\a\1\MaestroApplication\projectartifacts\ApplicationParameters\Int.xml' specified in the Azure Pipelines task.
Service fabric SDK version: 4.1.428.9590.
##[error]The environment variable 'ASPNETCORE_ENVIRONMENT' specified in EnvironmentOverrides for CodePackage 'Code', service manifest 'CoreHealthMonitorPkg' is not valid. Please check if EnvironmentVariables for the CodePackage contain a variable by this name.
FileName: C:\Users\VssAdministrator\AppData\Local\Temp\TestApplicationPackage_2595314239\j5y0yahs.szc\applicationpackage\ApplicationManifest.xml
```

https://dnceng.visualstudio.com/internal/_build/results?buildId=766241&view=logs&j=b277f1ae-44e7-5ce4-4818-00049b2c7654&t=79bc7191-4834-5309-e854-7f67053dbaa3

**Comment by ChadNedzlek** [comment]
God I had SF. :-|

**Comment by ChadNedzlek** [comment]
this should fix it. I can F5 locally. https://github.com/dotnet/arcade-services/pull/1343

**Pull request merged**